test/src/test/java/org/corfudb/infrastructure/log/StreamLogCompactionTest.java

```diff
@@ -0,0 +1,49 @@
+package org.corfudb.infrastructure.log;
+
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.AbstractCorfuTest;
+import org.corfudb.runtime.CorfuRuntime;
+import org.junit.Test;
+
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+
+@Slf4j
+public class StreamLogCompactionTest extends AbstractCorfuTest {
+
+    /**
+     * Test that task catch all possible exceptions and doesn't break scheduled executor
+     *
+     * @throws InterruptedException thread sleep
+     */
+    @Test
+    public void testCompaction() throws InterruptedException {
+        log.debug("Start log compaction test");
+
+        final int timeout = 35;
+        final int initialDelay = 10;
+        final int period = 10;
+
+        StreamLog streamLog = mock(StreamLog.class);
+        doThrow(new RuntimeException("err")).when(streamLog).compact();
+
+        StreamLogCompaction compaction = new StreamLogCompaction(
+                streamLog, initialDelay, period, TimeUnit.MILLISECONDS, Duration.ofSeconds(1)
+        );
+
+        TimeUnit.MILLISECONDS.sleep(timeout);
+        compaction.shutdown();
+
+        long gcCounter = CorfuRuntime.getDefaultMetrics()
+                .getCounters()
+                .get(StreamLogCompaction.STREAM_COMPACT_METRIC)
+                .getCount();
+
+        final int expectedGcCounter = 2;
+        assertTrue(gcCounter >= expectedGcCounter);
```

Comment:
nit: You can use `assertThat().isGreaterThanOrEqualTo()`. This gives the values as output in case of failure rather than just assertionError.